lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -167,9 +170,13 @@ def _match_installed_flat_name(module, binary, name, method):
     command = "{0} list --{1} --app".format(binary, method)
     output = _flatpak_command(module, False, command)
     parsed_name = _parse_flatpak_name(name)
-    for row in output.split('\n'):
-        if parsed_name.lower() in row.lower():
-            return row.split()[0]
+    # Convert name to byte-like object (For Python 2/3 compatibility)
+    byte_parsed_name = codecs.encode(parsed_name, 'utf-8')
+    for line in output.splitlines():
+        if byte_parsed_name.lower() in line.lower():
+            matched_name = line.split()[0]
+            # decode the byte-like object into a string
+            return codecs.decode(matched_name, 'utf-8')
```

Comment:
Use ansible. module_utils.to_native or to_text here.  To_Native converts to the str type (which is bytes on Python2 and text on python3). To_text converts to text type (Unicode on Python2.  Str on python 3).  Which you choose depends on what the rest of the module uses..

Reply:
See my other reply.